### ISSUE TITLE
Issue with version comparison

### ISSUE THREAD

**Issue by colececil** [colececil/familiar.sh#16]
There is a bug in the `compareVersionStrings` function in `version.go`. It currently splits versions into parts, where each part is separated by a `.` in the version string. Then it goes through each part and makes a comparison until it finds a part that is greater in one version.

The issue is that it's just using string comparison for each part. An example where this fails is when comparing version `1.2.6.863.ge7902f05` with version `1.2.10.760.g52970952`. It should find in the third part that `10` is greater than `6`, but that's not the case with regular string comparison. Instead, it thinks `6` is greater.

**Comment by colececil** [comment]
Another question here - do I need to do anything special with hyphens? For instance, semver shows pre-release versions with strings like "1.4.3-develop.1". I'm not sure if Familiar.sh will be used to install pre-release versions of anything, but other version strings out there might use hyphens for other things.

**Comment by colececil** [comment]
This is now fixed, and I wrote unit tests for the version comparison code. These are the version comparison rules I settled on;

1. Break up the strings into parts, considering any non-alphanumeric character to be a separator. Consider all separators to be equivalent. (We don't care which separator was used - only that a separator was used.)
2. Starting from the left, compare the respective parts from both versions until we find a part that is greater or smaller. If we reach the end and haven't found one, then the versions are equal.
    1. If one version has more parts than the other, consider missing parts to be "0".
    2. For each part, use numeric comparison if that part for both versions consists only of numbers. Otherwise, use lexicographic comparison.
